python/ray/data/dataset.py

```diff
@@ -2861,19 +2860,27 @@ def __iter__(self):
             )
         return pipe
 
-    def fully_executed(self) -> "Dataset[T]":
+    def fully_executed(self, preserve_original: bool = True) -> "Dataset[T]":
         """Force full evaluation of the blocks of this dataset.
 
         This can be used to read all blocks into memory. By default, Datasets
         doesn't read blocks from the datasource until the first transform.
 
+        Args:
+            preserve_original: Whether the original unexecuted dataset should be
```

Comment:
What is "original dataset" for a dataset? (I think users may have same question if they read this API)

Reply:
Ah I thought that when calling `d2 = ds.fully_executed(preserve_original=False)`, `original` referring to `ds` would be obvious. 🤔

Reply:
I see. I was having a chain of operations in mind, in that case it's unclear which one of them should be thought as "original" (maybe the first one? maybe each intermediate ones? or others?).